hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/storage/ContainerProtocolCalls.java

```diff
@@ -291,7 +292,10 @@ public static XceiverClientReply putBlockAsync(XceiverClientSpi xceiverClient,
       throws IOException, InterruptedException, ExecutionException {
     final ContainerCommandRequestProto request = getPutBlockRequest(
         xceiverClient.getPipeline(), containerBlockData, eof, tokenString);
-    return xceiverClient.sendCommandAsync(request);
+    // TODO: Make only putBlock triggered from hsync ALL_COMMITTED.
+    //  While keeping the ones triggered from write() and others at MAJORITY
+    //  and call regular watchForCommit in order to reduce latency for write()s.
```

Comment:
consider the case where PutBlock is piggybacked in a WriteChunk too.

Reply:
Yes this is considered. Currently a TODO just down below in `writeChunkAsync()`.

When PutBlock is piggybacked in WriteChunk, `putBlockAsync()` won't be called.